pkg/cmd/server/origin/openshift_apiserver.go

```diff
@@ -185,8 +186,24 @@ func (c *OpenshiftAPIConfig) SkipComplete() completedConfig {
 	return completedConfig{c}
 }
 
+func (c completedConfig) newTemplateConfig() *templateapiserver.TemplateConfig {
+	return &templateapiserver.TemplateConfig{
+		GenericConfig:       c.GenericConfig,
+		AuthorizationClient: c.KubeClientInternal.Authorization(),
+		Codecs:              kapi.Codecs,
+		Registry:            kapi.Registry,
+		Scheme:              kapi.Scheme,
+	}
+}
```

Comment:
this seems like a strange place to have template specific code/logic, what's the design here?